### PR TITLE
Fixes #5633 - Allow to configure HttpClient request authority.

### LINKED ISSUE #5633: Allow to configure HttpClient request authority
**Jetty version**
9.4.x

**Description**
When `HttpClient` is used in a proxy, it may connect to backend servers using a destination that is different from the authority that the remote client used.

For example, the `HttpClient` in the proxy may choose to connect to the backend server at `1.2.3.4:5678` while the client sent the request to `foo.bar.com:80`.

Would be great to be able to specify the `authority` of the request independently from the destination, along these lines:

```java
httpClient.newRequest("1.2.3.4", 5678)
  .authority("foo.bar.com:80")
  ...
```

### PR DESCRIPTION
**Pull request by sbordet**
Introduced HttpDestination.send(Request, Response.CompleteListener) to send a request using the given destination.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Review by gregw — APPROVED** [review]
lots of methods called send.... hard to follow.... but the unit test is good, so ultimately LGTM

**Review by joakime — APPROVED** [review]
LGTM
